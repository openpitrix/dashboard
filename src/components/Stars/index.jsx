import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { Icon } from 'components/Base';

import styles from './index.scss';

const stars = [1, 2, 3, 4, 5];

export default class Stars extends PureComponent {
  static propTypes = {
    className: PropTypes.string,
    starTotal: PropTypes.number
  };

  static defaultProps = {
    starTotal: 5
  };

  render() {
    const { className, starTotal } = this.props;

    return (
      <div className={classnames(styles.stars, className)}>
        {stars.map(star => (
          <Icon
            key={star}
            name="star"
            size={16}
            type="dark"
            className={classnames({
              [styles.yellow]: starTotal >= star
            })}
          />
        ))}
      </div>
    );
  }
}
