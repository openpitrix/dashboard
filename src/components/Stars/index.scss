@import '~scss/vars';

.stars {
  display: inline-block;
  .yellow svg {
    --primary-color: #{$Y75};
    --secondary-color: #{$Y75};
  }
}
