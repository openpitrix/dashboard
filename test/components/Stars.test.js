import Stars from 'components/Stars';

describe('Component Stars', () => {
  it('basic render', () => {
    const wrapper = render(<Stars starTotal={5}/>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
