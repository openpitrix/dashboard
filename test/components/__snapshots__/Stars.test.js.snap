// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Stars basic render 1`] = `
<div
  class="stars"
>
  <span
    class="icon icon-star yellow"
    style="width:16px;height:16px"
  >
    <svg
      class="qicon qicon-star qicon-dark"
    >
      <use
        href="#qui-star"
      />
    </svg>
  </span>
  <span
    class="icon icon-star yellow"
    style="width:16px;height:16px"
  >
    <svg
      class="qicon qicon-star qicon-dark"
    >
      <use
        href="#qui-star"
      />
    </svg>
  </span>
  <span
    class="icon icon-star yellow"
    style="width:16px;height:16px"
  >
    <svg
      class="qicon qicon-star qicon-dark"
    >
      <use
        href="#qui-star"
      />
    </svg>
  </span>
  <span
    class="icon icon-star yellow"
    style="width:16px;height:16px"
  >
    <svg
      class="qicon qicon-star qicon-dark"
    >
      <use
        href="#qui-star"
      />
    </svg>
  </span>
  <span
    class="icon icon-star yellow"
    style="width:16px;height:16px"
  >
    <svg
      class="qicon qicon-star qicon-dark"
    >
      <use
        href="#qui-star"
      />
    </svg>
  </span>
</div>
`;
